src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSubworkflowRedeclarationInspection.kt

```diff
@@ -16,7 +17,9 @@ class SmkSubworkflowRedeclarationInspection : SnakemakeInspection() {
         override fun visitSMKSubworkflow(subworkflow: SmkSubworkflow) {
             val nameNode = subworkflow.getNameNode()?.psi ?: return
```

Comment:
Use `subforkflow.name` instead of `subworkflow.getNameNode()?.psi.text`, better not to expose ASTNode related interface out of PsiElements. PsiElements - are containters for high level inteface for iteracting parsed elements with other code. ASTNodes - are too low-level API

Reply:
Probably better to replace `collectSubworkflows(): List<Pair<String, PsiElement>>` with `collectSubworkflows(): List<Pair<String, SmkSubworkflow>>` and do the same for rules.

Reply:
And check smth like this `subworkflow !== subworkflows.findLast { it.name = name }.second`